### DISCUSSION TITLE
Pay-by-Check Payment Type

### DISCUSSION THREAD

**Discussion by aelfric** [alfio-event/alf.io#1411] [Dev]
I'm a Java developer with Spring experience, but I'm also affiliated with a non-profit that is considering this software for an event this summer. Many of our attendees prefer to pay by check. I'm interested in contributing a payment method that behaves similar to the bank transfer option but with different end-user messaging. I haven't looked through any of the code yet, but is this something practical for a newcomer to the project to work on? If so, any guidance where to begin?

**Comment by cbellone** [comment]
Hi @aelfric ,

first of all: thanks for considering alf.io! 

> Many of our attendees prefer to pay by check. I'm interested in contributing a payment method that behaves similar to the bank transfer option but with different end-user messaging

I would suggest you take a look at our "Internationalization" configuration first 

![image](https://github.com/user-attachments/assets/2ca6657a-d222-4d1e-ae86-96576102b7bd)

there you can modify all the  messages displayed on the end-users form, for example you can change `reservation-page.offline` to be displayed as `Pay By Check`.

If that wouldn't be enough for your use case, we'd be glad if you could contribute a generic "custom offline payment method", which would also work for other "offline" payment methods, like https://github.com/alfio-event/alf.io/issues/1384

let me know, thanks
Celestino

**Reply by aelfric** [reply]
That seems like a good start. I'll take a closer look at all those settings and reply back if I run into any problems.

**Comment by EDDYYUNGMEDIA** [comment]
> I'm a Java developer with Spring experience, but I'm also affiliated with a non-profit that is considering this software for an event this summer. Many of our attendees prefer to pay by check. I'm interested in contributing a payment method that behaves similar to the bank transfer option but with different end-user messaging. I haven't looked through any of the code yet, but is this something practical for a newcomer to the project to work on? If so, any guidance where to begin?

Interested
